routes/memberProjects.js

```diff
@@ -88,4 +89,53 @@ router.get("/by-key/:associationId", async (req, res) => {
   });
 });
 
+router.get("/impersonate/:associationId", async (req, res) => {
+  const key = await req.ctx.db.collection("keys").findOne({
+    [req.associationType]: {$eq: req.params.associationId},
+  });
+
+  if (!key) {
+    res.status(404).json({message: "Not found"});
+    return;
+  }
+
+  const userDocument = await req.ctx.db.collection("users").findOne({
+    id: {$eq: key.userId},
+    disabled: {$ne: true},
+  });
+  if (!userDocument) {
+    res.status(404).json({message: "User not found or disabled"});
+    return;
+  }
+
+  let user;
+  try {
+    user = await findUser(key.userId);
+  } catch (err) {
+    res.status(500).json({message: "Internal server error"});
+    return;
+  }
+
+  const response = {};
+  for (const attribute of user.attributes) {
+    if (attribute.type == "jpegPhoto") {
+      response[attribute.type] = attribute._vals[0].toString("base64");
+    } else {
+      const values = attribute._vals.map((value) => value.toString("utf8"));
+      if (ARRAYS.has(attribute.type)) {
+        response[attribute.type] = values;
+      } else {
+        if (values.length > 1) {
+          console.warn(`${attribute.type} has many values!!`);
+        }
+        response[attribute.type] = values.join(",");
+      }
+    }
+  }
+
+  const uid = response["uid"];
+
+  res.json(await impersonate.getImpersonationToken(uid));
```

Comment:
It would be nice if we added `ipaUniqueID` to the response too, I really want to encourage people to use that attribute where possible